crates/sui-types/src/effects/effects_v2.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::base_types::{EpochId, ObjectDigest, ObjectRef, TransactionDigest};
+use crate::digests::TransactionEventsDigest;
+use crate::effects::{TransactionEffectsAPI, TransactionEffectsDebugSummary};
+use crate::execution_status::ExecutionStatus;
+use crate::gas::GasCostSummary;
+use crate::object::Owner;
+use crate::storage::{DeleteKind, WriteKind};
+use crate::{ObjectID, SequenceNumber};
+use serde::{Deserialize, Serialize};
+use std::cell::OnceCell;
+use std::collections::HashSet;
+
+/// The response from processing a transaction or a certified transaction
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub struct TransactionEffectsV2 {
+    /// The status of the execution
+    status: ExecutionStatus,
+    /// The epoch when this transaction was executed.
+    executed_epoch: EpochId,
+    gas_used: GasCostSummary,
```

Comment:
I seemed to remember some folks asking for a more detailed gas breakdown than what we currently had. If i remember this correctly maybe it would be a good idea to look into improving this as well?

Reply:
I don't know. I would probably focus on object changes in v2. 
But if there is a reasonable proposal for including more gas breakdown we can discuss. My concern is that a detailed breakdown can be large and pins us to the implementation details (we cannot and shouldn't iterate effects too often).
@dariorussi

Reply:
I think the ideal place for a more detailed gas breakdown is in something like dev-inspect and/or dry-run.